### ISSUE TITLE
Workspace Not Found error when copying Insomnia files

### ISSUE THREAD

**Issue by designbyadrian** [ArchGPT/insomnium#44]
### Expected Behavior

Application starts with my settings from Insomnia

### Actual Behavior

Application hopens with this window:

<img width="1392" alt="image" src="https://github.com/ArchGPT/insomnium/assets/2064558/eea5816b-33a7-4128-848e-774c2cc4f9b4">

Error: Workspace not found at _t (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:68:7551) at Module.Ot (file:///Applications/Insomnium.app/Contents/Resources/app.asar/workspace-c2145993.js:1:491) at async kl (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:50:3691) at async Promise.all (index 1) at async Ie (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:21936) at async He (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:16864) at async le (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:15041) at async s1 (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:21835) at async He (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:17112) at async le (file:///Applications/Insomnium.app/Contents/Resources/app.asar/index-e5e8515f.js:49:15041)

### Reproduction Steps

* Download Insomnium
* Open Insomnium
* Close Insomnium
* Run the command from the [migration guide](https://archgpt.dev/insomnium/migration-guide): `cp -r ~/Library/Application\ Support/Insomnia/*.db ~/Library/Application\ Support/Insomnium/`

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/ArchGPT/insomnium/issues) for this problem.

### Insomnium Version

0.2.1-b

### What operating system are you using?

macOS

### Operating System Version

macOS Sonoma 14.0

### Installation method

Downloaded from GitHub repository

**Comment by archywillhe** [comment]
hey thanks for posting this bug; 

did it work after you reload the app? 

(btw your data should be still inside `~/Library/Application\ Support/Insomnia/*.db` )

**Comment by archywillhe** [comment]
hey thanks for posting this bug; 

I think it should work after you click on "Try to reload this app" button. 

a fix will be out in an hour

**Comment by archywillhe** [comment]
Were you coming from Insomnia 8.0 or above? 

Or were you on the older version of the Insomnia?

**Comment by designbyadrian** [comment]
@archywillhe From Insomnia 8. I wish I had copied the data from the older version, but I couldn't forsee what a complete mess the update would be.

After clicking "Try to reload", the app does indeed start and work properly, but has no content.

**Comment by archywillhe** [comment]
Yeah.. This problem arises because the data structure implemented in `Insomnia 8` is quite from its pervious version, and so it messes the exsting .db files in `~/Library/Application\ Support/Insomnia`. 

(This is why after you copied over to the `Insomnium` folder, this crash would arise and after reload you would be left with an empty "Scratch Pad" collection.) Insomnium is a fork based on [v2023.5.8](https://github.com/ArchGPT/insomnia) and it's quite cumbersome to implemented a direct fix to this. 

To get your data back, the easiest way now is to download [Insomnia 2023.5.8](https://github.com/Kong/insomnia/releases/tag/core%402023.5.8) and reopen the Insomnia and the data should be back. (As long as you don't upgrade it; the data should be there.) And then close Insomnia and peform `cp -r ~/Library/Application\ Support/Insomnia/*.db ~/Library/Application\ Support/Insomnium/` again. Now if you open Insomnium, you should see everything intact. 

In short `Insomnia 8` -> reinstall/downgrade back to `Insomnia 2023.5.8` -> open `Insomnia 2023.5.8` to fix this, and now the migration to `Insomnium` should work.

**Comment by archywillhe** [comment]
Yeah.. This problem arises because the data structure implemented in `Insomnia 8` is substantially different from its pervious version, and so it messes the exsting .db files in `~/Library/Application\ Support/Insomnia`. 

(This is why after you `cp` over to the `Insomnium` folder, this crash would arise and after reload you would be left with an empty "Scratch Pad" collection.) Insomnium is a fork based on [v2023.5.8](https://github.com/ArchGPT/insomnia) and it parses data the same way Insomnia 2023.5.8 does.

To get your data back, the easiest way now is to download [Insomnia 2023.5.8](https://github.com/Kong/insomnia/releases/tag/core%402023.5.8) and reopen the Insomnia and the data should be back. (As long as you don't upgrade it; the data should be there.) And then close Insomnia and peform `cp -r ~/Library/Application\ Support/Insomnia/*.db ~/Library/Application\ Support/Insomnium/` again. Now if you open Insomnium, you should see everything intact. 

In short `Insomnia 8` -> reinstall/downgrade back to `Insomnia 2023.5.8` -> open `Insomnia 2023.5.8` to fix this, and now the migration to `Insomnium` should work.
